src/cairoWriter/writers/literalWriter.ts

```diff
@@ -1,22 +1,21 @@
 import { ASTWriter, Literal, LiteralKind, SrcDesc } from 'solc-typed-ast';
 import { TranspileFailedError } from '../../utils/errors';
-import { divmod, primitiveTypeToCairo } from '../../utils/utils';
+import { primitiveTypeToCairo, CairoPrimitiveIntType } from '../../utils/utils';
 import { CairoASTNodeWriter } from '../base';
 
 export class LiteralWriter extends CairoASTNodeWriter {
   writeInner(node: Literal, _: ASTWriter): SrcDesc {
     switch (node.kind) {
       case LiteralKind.Number:
-        switch (primitiveTypeToCairo(node.typeString)) {
-          case 'Uint256': {
-            const [high, low] = divmod(BigInt(node.value), BigInt(Math.pow(2, 128)));
-            return [`Uint256(low=${low}, high=${high})`];
-          }
-          case 'felt':
-            return [node.value];
-          default:
-            throw new TranspileFailedError('Attempted to write unexpected cairo type');
+        if (
+          (primitiveTypeToCairo(node.typeString) as CairoPrimitiveIntType) ||
+          primitiveTypeToCairo(node.typeString) === 'felt'
```

Comment:
Can we use typenodes instead? We should avoid using typestrings due to solc-typed-ast update

Reply:
we're defining correspondence between sol int type typestring to cairo int type typestring, the function `primitiveTypeToCairo` takes for examples `uint24` as an input and returns `u24` as a output